Add better support for parsing vCard 2.1 files

schmod opened the pull request:
Adds better support for parsing old vCard 2.1 files (#1).  Outlook and several other email clients still produce files in this format, so they're regrettably still fairly common in the wild.
- ADR fields are now supported.
  - LABELs still need to be implemented.
- QUOTED-PRINTABLE values are now correctly identified and decoded
- Supports vCard 2.1's boolean PREF attribute.
- Instead of discarding unknown attributes, assume that they belong to 'TYPE,' which they usually are in vC 2.1 files.
  - We could do this more intelligently by checking against the table in section 10.3.4 of the RFC.

schmod committed:
Add support for 'ADR' field.
TODO: Also handle 'LABEL' field (which lives inside 'ADR' in vC 3.0+)

schmod committed:
Several changes to support vCard 2.1 files as best as we can.

Inculdes support for reading multiline quoted-printables.
TODO: Refactor for readability, and actually decode the QPs.

Unknown attributes are inferred to be 'TYPE' -- if we want to do this
rigorously, we can validate more thoroughly against the spec.
Section 10.3.4 of the RFC gives valid values for 'TYPE'

schmod committed:
Decode quoted-printables.

nilclass commented:
Looks great!
Before I merge, could you add comments to all the code that is only there for compatibility with older files?

Considering unknown attributes as "TYPE" attributes sounds reasonable. It's better than throwing the data away.

schmod committed:
Comment code to indicate vC 2.1-specific hacks

schmod committed:
Clarify one of those comments

Q-P linesplitting is colloquially referred to as "folding," so that's
what we'll call it.

schmod commented:
Just pushed a commit with some more comments to indicate the stuff that was added for vCard 2.1 compatibility.  

Another small change that I forgot to mention above is that attributes are now parsed into an array, rather than a comma-delimited list, ie:

```
TYPE: 'CELL,WORK'
```

would now be:

```
TYPE: ['CELL','WORK']
```

This matches the jCard spec, and just seems a whole lot easier to work with.

nilclass commented:
Cool, thanks! Sorry for taking so long to merge, kind of lost track of this PR :)

schmod commented:
No worries.  Thanks for considering it.  I'd love to implement more of the spec, so I may end up back here in the future...

Would you mind running the makefile to create a new release?

nilclass commented:
good point. did that.
